python-main.js

```diff
@@ -1212,23 +1255,11 @@ function web_editor(config) {
             // var timeTaken = (new Date().getTime() - startTime) / (1000 * 60);
             // console.log("Time taken to flash: " + Math.floor(timeTaken) + " minutes, "
             //           + Math.ceil((timeTaken - Math.floor(timeTaken)) * 60) + " seconds");
-        })
-        .catch(function(err) {
-            console.log("Error flashing: " + err);
-            $("#flashing-overlay-container").css("display", "flex");
-            $("#webusb-flashing-progress").css("display", "none");
-
-            // If micro:bit does not support dapjs
-            if (err.message === "No valid interfaces found.") {
-                $("#flashing-overlay-error").html('<div>' + err + '</div><div>'+ config["translate"]["webusb"]["err"]["update-req"] +'</div><a href="#" onclick="flashErrorClose()">'+ config["translate"]["webusb"]["close"] +'</a>');
-                return;
-            } else if(err.message === "Unable to claim interface.") {
-                $("#flashing-overlay-error").html('<div>'+ config["translate"]["webusb"]["err"]["clear-connect"] +'</div><a href="#" onclick="flashErrorClose()">'+ config["translate"]["webusb"]["close"] +'</a>');
-                return;
-            }
 
-            $("#flashing-overlay-error").html('<div>' + err + '</div><div>'+ config["translate"]["webusb"]["err"]["restart-microbit"] +'</div><a href="#" onclick="flashErrorClose()">'+ config["translate"]["webusb"]["close"] +'</a>');
-        });
+            // Remove event listener
+            window.removeEventListener("unhandledrejection", webusbErrorHandler);
```

Comment:
Is Promise.finally() a good alternative to ensure this listener is always removed?
I don't know, so that's a question more than a suggestion.

Reply:
I think you're right, I'll give it a go